From baa1c65c26f0eec9579664936d7bc29c3be51bde Mon Sep 17 00:00:00 2001
From: Jaroslav Mracek <jmracek@redhat.com>
Date: Tue, 3 Sep 2019 11:01:51 +0200
Subject: [PATCH] Keep installed packages in upgrade job (RhBug:1728252,1644241,1741381)

In combination with marking of job as TARGETED it prevents from
reinstalling of modified packages with same NEVRA.

https://bugzilla.redhat.com/show_bug.cgi?id=1728252
https://bugzilla.redhat.com/show_bug.cgi?id=1644241
https://bugzilla.redhat.com/show_bug.cgi?id=1741381

Closes: #1474
Approved by: m-blaha
---
 dnf.spec                  | 6 +++++-
 dnf/base.py               | 3 ---
 dnf/module/module_base.py | 2 +-
 3 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/dnf.spec b/dnf.spec
index 56fdea7..2e60265 100644
--- a/dnf.spec
+++ b/dnf.spec
@@ -79,7 +79,7 @@
 It supports RPMs, modules and comps groups & environments.
 
 Name:           dnf
-Version:        4.2.8
+Version:        4.2.11
 Release:        1%{?dist}
 Summary:        %{pkg_summary}
 # For a breakdown of the licensing, see PACKAGE-LICENSING
@@ -502,6 +502,10 @@ ln -sr  %{buildroot}%{confdir}/vars %{buildroot}%{_sysconfdir}/yum/vars
 %endif
 
 %changelog
+* Tue Aug 13 2019 Pavla Kratochvilova <pkratoch@redhat.com> - 4.2.9-1
+- Prevent printing empty Error Summary (RhBug: 1690414)
+- [doc] Add user_agent and countme options
+
 * Tue Aug 06 2019 Pavla Kratochvilova <pkratoch@redhat.com> - 4.2.8-1
 - Enhance synchronization of rpm transaction to swdb
 - Accept multiple specs in repoquery options (RhBug:1667898)
diff --git a/dnf/base.py b/dnf/base.py
index b0e1313..881e370 100644
--- a/dnf/base.py
+++ b/dnf/base.py
@@ -1963,9 +1963,6 @@ class Base(object):
                 obsoletes=q.installed().union(q.upgrades()))
             # add obsoletes into transaction
             q = q.union(obsoletes)
-        # provide only available packages to solver otherwise selection of available
-        # possibilities will be ignored
-        q = q.available()
         if reponame is not None:
             q.filterm(reponame=reponame)
         q = self._merge_update_filters(q, pkg_spec=pkg_spec)
diff --git a/dnf/module/module_base.py b/dnf/module/module_base.py
index c86eabe..0fe1179 100644
--- a/dnf/module/module_base.py
+++ b/dnf/module/module_base.py
@@ -214,7 +214,7 @@ class ModuleBase(object):
 
             if not upgrade_package_set:
                 logger.error(_("Unable to match profile in argument {}").format(spec))
-            query = self.base.sack.query().available().filterm(name=upgrade_package_set)
+            query = self.base.sack.query().filterm(name=upgrade_package_set)
             if query:
                 sltr = dnf.selector.Selector(self.base.sack)
                 sltr.set(pkg=query)
--
libgit2 0.28.2

